nessie/src/main/java/org/apache/iceberg/nessie/NessieIcebergClient.java

```diff
@@ -540,4 +604,35 @@ public void close() {
       api.close();
     }
   }
+
+  private void commitRetry(String message, Operation... ops)
+      throws BaseNessieClientServerException {
+
+    CommitMultipleOperationsBuilder commitBuilder =
+        api.commitMultipleOperations()
+            .commitMeta(NessieUtil.buildCommitMetadata(message, catalogOptions))
+            .operations(Arrays.asList(ops));
+
+    Tasks.foreach(commitBuilder)
+        .retry(5)
+        .stopRetryOn(NessieReferenceNotFoundException.class, NessieReferenceConflictException.class)
+        .throwFailureWhenFinished()
+        .onFailure((o, exception) -> refresh())
```

Comment:
I'm still not sure this refresh is meaningful. A refresh could be helpful to replace out-of-date local data, but if that is the case, we'd get a conflict and stop retrying anyway.

I know this refresh was present before, so it is fine to keep the same behaviour for now, but we should probably revisit this logic in another PR. WDYT?